pkg/storageincentives/agent.go

```diff
@@ -369,6 +369,9 @@ func (a *Agent) handleClaim(ctx context.Context, round uint64) (bool, error) {
 }
 
 func (a *Agent) handleSample(ctx context.Context, round uint64) (bool, error) {
+
+	a.state.SetLastPlayedRound(round)
+
```

Comment:
I see, correct me if I'm wrong but if its at the end of handleSample func doesn't that mean it will be updated only when the node's neighbourhood is selected?( `a.contract.isPlaying`) From #3979 i believe the "lastPlayedRound" referred to the previous round not necessarily one that the node's neighbourhood participated in hence the distinction with the "lastSelectedRound".

Reply:
I thought lastSelectedRound is when the neighborhood was selected.  lastPlayedRound would be one where the node actually submitted (or possibly calculated) a sample hash meaning that the node actually played the round.

Reply:
You're right, will address